src/utils/validation/basic_data/addEmployeeValidation.ts

```diff
@@ -70,9 +70,12 @@ export const userValidationSchema = Yup.object<YupShape<FlatCristinPerson>>({
     .matches(nameRegexp, employeeErrorMessage.lastNameInvalidFormat)
     .min(2, employeeErrorMessage.lastNameRequired)
     .required(employeeErrorMessage.lastNameRequired),
+  nvi: Yup.object(),
   nationalId: Yup.string()
-    .matches(/^\d{11}$/, employeeErrorMessage.nationalIdInvalidFormat)
-    .required(employeeErrorMessage.nationalIdInvalid),
+    .when('nvi', ([nvi], schema) =>
+      nvi.verifiedBy.id === '' ? schema.required(employeeErrorMessage.nationalIdInvalid) : schema.optional()
+    )
+    .matches(/^\d{11}$/, employeeErrorMessage.nationalIdInvalidFormat),
```

Comment:
burde ikke format-sjekken også ligge her?
```suggestion
      nvi.verifiedBy.id === '' ? schema.matches(/^\d{11}$/, employeeErrorMessage.nationalIdInvalidFormat).required(employeeErrorMessage.nationalIdInvalid) : schema.optional()
    )
```

Reply:
ev. kunne den kanskje vært før `when` også

Reply:
Blir vel riktig validert nå?
![image](https://github.com/BIBSYSDEV/NVA-Frontend/assets/53341294/e54b0d4e-0f23-4502-8460-53beb35cf1db)

Reply:
cmiiw, men tror den sjekker om den matcher regexen før den sjekker om den er optional eller ikke, så da validerer den fortsatt om man huker av checkboxen og den mister fokus.

Reply:
primært for optimalisering tenkte jeg. 
Ikke at dette er noe brukere kommer til å merke, men tror at slik det er nå vil vi kjøre regex-match også når vi vet det ikke er et f.nummer der, og da heller ikke bryr oss om hvordan formatet er. Så litt bedre praksis å validere ting bare når vi faktisk er interessert i utfallet